lobster/tools/core/online_report.py

```diff
@@ -32,6 +32,48 @@ class Parse_Error(Exception):
     pass
 
 
+def is_git_main_module(path):
+    return os.path.isdir(os.path.join(path, ".git"))
+
+
+def find_repo_main_root(file_path):
```

Comment:
I did a bit of research. It seems the most robust way is to do the following:
1. first check if the folder is inside a git submodule using `--show-superproject-working-tree`
2. if not, then check if it is inside a git working tree using `--show-toplevel`

This is code that our AI generated:
```python
import os
import subprocess

def is_dir_in_git_submodule(directory):
    """
    Checks if a given directory is nested inside a Git submodule.
    
    Args:
        directory (str): The path to the directory to check.
    
    Returns:
        bool: True if the directory is inside a Git submodule, False otherwise.
    """
    try:
        # Check if the directory is part of a Git submodule
        result = subprocess.run(['git', 'rev-parse', '--show-superproject-working-tree'], cwd=directory, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
        if result.returncode == 0 and result.stdout.strip():
            return True
        else:
            return False
    except (subprocess.CalledProcessError, OSError):
        return False

def is_dir_in_git_main_module(directory):
    """
    Checks if a given directory is nested inside a Git main module.
    
    Args:
        directory (str): The path to the directory to check.
    
    Returns:
        bool: True if the directory is inside a Git main module, False otherwise.
    """
    try:
        # Check if the directory is part of a Git main module
        result = subprocess.run(['git', 'rev-parse', '--show-toplevel'], cwd=directory, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
        if result.returncode == 0 and result.stdout.strip():
            return True
        else:
            return False
    except (subprocess.CalledProcessError, OSError):
        return False
```